packages/libs/client/src/pact.ts

```diff
@@ -277,15 +281,24 @@ export class PactCommand
    * Checks if a transaction succeeded or failed by polling the apiHost at
    * a given interval. Times out if it takes too long.
    * @param apiHost - the chainweb host where to send the transaction to
-   * @param interval - the amount of time in ms between the api calls
-   * @param timeout - the total time in ms after this function will time out
+   * @param interval - the amount of time in ms between the api calls (optional)
+   * @param timeout - the total time in ms after this function will time out (optional)
+   * @param onPoll - a function that gets called after each poll with the IPollResult as a parameter (optional)
    * @alpha
    */
-  public callPollUntilTimeout(
+  public pollUntil(
     apiHost: string,
-    interval: number = 5000,
-    timeout: number = 60000,
+    options: {
+      interval?: number;
+      timeout?: number;
+      onPoll?: (result: IPollResponse) => void;
+    },
```

Comment:
Would it be an idea to destruct options to the arguments and do onPoll first. 
Then we can set defaults there, and get rid of the ones on line 301.

Especially with the 5 seconds interval and 3 minutes timeout as defaults, you wouldn't set them, but you _would_ want to set the onPoll almost always.

```suggestion
    onPoll?: (result: IPollResponse) => void;
    interval: number = 1000 * 5;  // 5 seconds
    timeout: number = 1000 * 60 * 3; // 3 minutes
```

Reply:
Decided to keep the `options` to be more flexible. Another benefit I see is that we don't need a breaking change if we ever decide to add another optional param.